package_test.go

```diff
@@ -1358,6 +1384,128 @@ AND    l.model_uuid = $JujuLeaseKey.model_uuid`,
+
+	v2 := int8(30)
+	o2 := int8(0)
+	stmt, err = sqlair.Prepare("SELECT id AS &int8 FROM person WHERE id = $int8")
+	c.Assert(err, IsNil)
+	err = db.Query(nil, stmt, v2).Get(&o2)
+	c.Assert(err, IsNil)
+	c.Assert(v2, Equals, o2)
+
+	v3 := int16(30)
+	o3 := int16(0)
+	stmt, err = sqlair.Prepare("SELECT id AS &int16 FROM person WHERE id = $int16")
+	c.Assert(err, IsNil)
+	err = db.Query(nil, stmt, v3).Get(&o3)
+	c.Assert(err, IsNil)
+	c.Assert(v3, Equals, o3)
+
+	v4 := int32(30)
+	o4 := int32(0)
+	stmt, err = sqlair.Prepare("SELECT id AS &int32 FROM person WHERE id = $int32")
+	c.Assert(err, IsNil)
+	err = db.Query(nil, stmt, v4).Get(&o4)
+	c.Assert(err, IsNil)
+	c.Assert(v4, Equals, o4)
+
+	v5 := int64(30)
+	o5 := int64(0)
+	stmt, err = sqlair.Prepare("SELECT id AS &int64 FROM person WHERE id = $int64")
+	c.Assert(err, IsNil)
+	err = db.Query(nil, stmt, v5).Get(&o5)
+	c.Assert(err, IsNil)
+	c.Assert(v5, Equals, o5)
+
+	v6 := uint(30)
+	o6 := uint(0)
+	stmt, err = sqlair.Prepare("SELECT id AS &uint FROM person WHERE id = $uint")
+	c.Assert(err, IsNil)
+	err = db.Query(nil, stmt, v6).Get(&o6)
+	c.Assert(err, IsNil)
+	c.Assert(v6, Equals, o6)
+
+	v7 := uint8(30)
+	o7 := uint8(0)
+	stmt, err = sqlair.Prepare("SELECT id AS &uint8 FROM person WHERE id = $uint8")
+	c.Assert(err, IsNil)
+	err = db.Query(nil, stmt, v7).Get(&o7)
+	c.Assert(err, IsNil)
+	c.Assert(v7, Equals, o7)
+
+	v8 := uint16(30)
+	o8 := uint16(0)
+	stmt, err = sqlair.Prepare("SELECT id AS &uint16 FROM person WHERE id = $uint16")
+	c.Assert(err, IsNil)
+	err = db.Query(nil, stmt, v8).Get(&o8)
+	c.Assert(err, IsNil)
+	c.Assert(v8, Equals, o8)
+
+	v9 := uint32(30)
+	o9 := uint32(0)
+	stmt, err = sqlair.Prepare("SELECT id AS &uint32 FROM person WHERE id = $uint32")
+	c.Assert(err, IsNil)
+	err = db.Query(nil, stmt, v9).Get(&o9)
+	c.Assert(err, IsNil)
+	c.Assert(v9, Equals, o9)
+
+	v10 := uint64(30)
+	o10 := uint64(0)
+	stmt, err = sqlair.Prepare("SELECT id AS &uint64 FROM person WHERE id = $uint64")
+	c.Assert(err, IsNil)
+	err = db.Query(nil, stmt, v10).Get(&o10)
+	c.Assert(err, IsNil)
+	c.Assert(v10, Equals, o10)
+
+	v11 := float32(30)
+	o11 := float32(0)
+	stmt, err = sqlair.Prepare("SELECT id AS &float32 FROM person WHERE id = $float32")
+	c.Assert(err, IsNil)
+	err = db.Query(nil, stmt, v11).Get(&o11)
+	c.Assert(err, IsNil)
+	c.Assert(v11, Equals, o11)
+
+	v12 := float64(30)
+	o12 := float64(0)
+	stmt, err = sqlair.Prepare("SELECT id AS &float64 FROM person WHERE id = $float64")
+	c.Assert(err, IsNil)
+	err = db.Query(nil, stmt, v12).Get(&o12)
+	c.Assert(err, IsNil)
+	c.Assert(v12, Equals, o12)
+
+	v13 := string("Fred")
+	o13 := string("")
+	stmt, err = sqlair.Prepare("SELECT name AS &string FROM person WHERE name = $string")
+	c.Assert(err, IsNil)
+	err = db.Query(nil, stmt, v13).Get(&o13)
+	c.Assert(err, IsNil)
+	c.Assert(v13, Equals, o13)
+
+	v14 := bool(true)
+	o14 := bool(false)
+	stmt, err = sqlair.Prepare("SELECT true AS &bool FROM person WHERE id > $bool")
```

Comment:
Why are we using `>` with bools?

Reply:
I've changed it to an `= 1` check. To be honest I'm not really sure how you would use a bool in a SQL query